src/cursor/cur_hs.c

```diff
@@ -1047,6 +1047,9 @@ __curhs_update(WT_CURSOR *cursor)
     hs_upd->durable_ts = hs_cursor->time_window.durable_start_ts;
     hs_upd->txnid = hs_cursor->time_window.start_txn;
 
+    WT_ASSERT(session,
```

Comment:
Do we want to remove this assert too?

Reply:
No need to remove this assertion. This API is called only from prepare commit/rollback. In those scenarios always the durable timestamp will be larger. While going through these changes, It is better to backport the WT-8425 into 5.0 as it fixes a history store bug.